apollo-federation/src/sources/connect/query_graph/fetch.rs

```diff
@@ -0,0 +1,284 @@
+        for (index, name) in ["Post", "User", "View"].into_iter().enumerate() {
+            let node_type = ObjectTypeDefinitionPosition {
+                type_name: Name::new(name).unwrap(),
+            };
+
+            let node = graph.add_node(FederatedQueryGraphNode::Concrete {
+                supergraph_type: node_type.clone(),
+                field_edges: IndexMap::new(),
+                source_exiting_edge: None,
+                source_id: source_id.clone(),
+                source_data: SourceFederatedConcreteQueryGraphNode::Connect(
+                    ConnectFederatedConcreteQueryGraphNode::SelectionRoot {
+                        subgraph_type: node_type.clone(),
+                        property_path: vec![Property::Field(name.to_lowercase().to_string())],
+                    },
+                ),
+            });
+
+            edges.push(graph.add_edge(
+                query,
+                node,
+                FederatedQueryGraphEdge::ConcreteField {
+                    supergraph_field: ObjectFieldDefinitionPosition {
+                        type_name: Name::new(name).unwrap(),
+                        field_name: Name::new(name.to_lowercase()).unwrap(),
+                    },
+                    self_conditions: None,
+                    source_id: source_id.clone(),
+                    source_data: None,
+                },
+            ));
+
+            // Optionally add the entrypoint
+            if index < entrypoints {
+                edges.push(graph.add_edge(
+                    query,
+                    node,
+                    FederatedQueryGraphEdge::SourceEntering {
+                        supergraph_type: node_type,
+                        self_conditions: None,
+                        tail_source_id: source_id.clone(),
+                        source_data: None,
+                    },
+                ));
+            }
+        }
+
+        let (entry, non_entry) = edges.into_iter().partition(|&edge_index| {
+            matches!(
+                graph.edge_weight(edge_index),
+                Some(FederatedQueryGraphEdge::SourceEntering { .. })
+            )
+        });
+        SetupInfo {
+            fetch_graph: ConnectFetchDependencyGraph,
+            query_graph: Arc::new(FederatedQueryGraph::with_graph(graph)),
+            source_id,
+            source_entry_edges: entry,
+            non_source_entry_edges: non_entry,
+        }
+    }
+
+    #[test]
+    fn it_handles_a_new_path() {
+        let SetupInfo {
+            fetch_graph,
+            query_graph,
+            source_entry_edges,
+            source_id,
+            ..
+        } = setup();
+
+        let path = fetch_graph
+            .new_path(query_graph, &[], source_entry_edges[0], None)
+            .unwrap();
+
+        assert_eq!(
+            path,
+            SourcePath::Connect(ConnectPath {
+                merge_at: Vec::new(),
+                source_entering_edge: source_entry_edges[0],
```

Comment:
tests that set up a variable and then assert that the actual result matches the variable smells to me. a reader of the test doesn't know what `source_entry_edges[0]` is, and there's always a slight possibility that you're `assert_eq!(None, None)`.

Reply:
I switched to using snapshots, which should help. For tests around PetGraph nodes, it'll still be a bit opaque as the indices are literally just numbers.